eth/db/journal.py

```diff
@@ -330,24 +349,42 @@ def commit(self, changeset_id: uuid.UUID) -> None:
 
         If this is the base changeset then all changes will be written to
         the underlying database and the Journal starts a new recording.
+        Typically, callers won't have access to the base changeset, because
+        it is dropped during .reset() which is called in JournalDB().
         """
         self._validate_changeset(changeset_id)
         journal_data = self.journal.commit_changeset(changeset_id)
 
         if self.journal.is_empty():
-            for key, value in journal_data.items():
-                if value is not DELETED_ENTRY:
-                    self.wrapped_db[key] = value  # type: ignore # value can only be bytes here
-                else:
-                    try:
-                        del self.wrapped_db[key]
-                    except KeyError:
-                        pass
-
             # Ensure the journal automatically restarts recording after
             # it has been persisted to the underlying db
             self.reset()
 
+            for key, value in journal_data.items():
+                try:
+                    if value is DELETED_ENTRY:
+                        del self.wrapped_db[key]
+                    elif value is ERASE_CREATED_ENTRY:
+                        pass
+                    else:
+                        self.wrapped_db[key] = value  # type: ignore # value can only be bytes here
```

Comment:
nit: instead of a `#type: ignore` I think a `cast(bytes, value)` would do the trick here.

Reply:
alternatively, `assert isinstance(value, bytes)` also makes mypy happy, for a performance penalty.

Reply:
Yeah, I'll do it for now. I seem to remember `JournalDB` being a place that was fairly sensitive to performance, but it could probably use a lot of performance love anyway, so I won't try to optimize without benchmarks. :)